patches/net/minecraft/world/entity/npc/VillagerTrades.java.patch

```diff
@@ -0,0 +1,22 @@
+--- a/net/minecraft/world/entity/npc/VillagerTrades.java
++++ b/net/minecraft/world/entity/npc/VillagerTrades.java
+@@ -1484,9 +_,6 @@
+         private final int villagerXp;
+ 
+         public EmeraldsForVillagerTypeItem(int p_35669_, int p_35670_, int p_35671_, Map<VillagerType, Item> p_35672_) {
+-            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_339515_ -> {
+-                throw new IllegalStateException("Missing trade for villager type: " + BuiltInRegistries.VILLAGER_TYPE.getKey(p_339515_));
+-            });
+             this.trades = p_35672_;
+             this.cost = p_35669_;
+             this.maxUses = p_35670_;
+@@ -1498,6 +_,9 @@
+         public MerchantOffer getOffer(Entity p_219685_, RandomSource p_219686_) {
+             if (p_219685_ instanceof VillagerDataHolder villagerdataholder) {
+                 ItemCost itemcost = new ItemCost(this.trades.get(villagerdataholder.getVillagerData().getType()), this.cost);
```

Comment:
This also needs to null check `trades#get`, no?

Reply:
Based on my testing apparently it doesn't but I will do more testing

Reply:
Actually yes the item needs to be checked for null

Reply:
Pushed fix